### DISCUSSION TITLE
Typing in v6

### DISCUSSION THREAD

**Discussion by ghost** [fabricjs/fabric.js#8798] [General]
Noting some typing issues in v6, will make a bug report if I have time for a reproducible repo 😅

**Comment by ghost** [comment]
🟢  `TDataUrlOptions` is not exported. Workaround:

```ts
type DataURLOptions = Exclude<
  Parameters<fabric.Canvas["toDataURL"]>[0],
  undefined
>;
```

**Comment by ghost** [comment]
🔴 `ImageFormat` is not exported. This is hard to work around because it's an enum. Doing this:

```ts
type ImageFormat = Parameters<typeof fabric.util["toDataURL"]>[2]
```

won't help because it only gives us a type of number, not an enum that we can use as values. Closest we can do is:

```ts
const FORMAT_PNG = "png" as DataURLOptions["format"];
```

**Comment by ghost** [comment]
🔴 `Object.toDataURL` has an `any` type. However it has pretty good JSDoc (but not sure if this is updated)

<details>
<summary>Screenshot</summary>
<img width="1264" alt="image" src="https://user-images.githubusercontent.com/122243953/226787931-e051d804-d27a-46aa-ad53-a1b3656bba3b.png">
</details>

As a workaround, the option here looks similar to `Canvas.toDataURL`, which is typed

**Comment by ShaMan123** [comment]
This is a known issue
We will address it
